pulp_plugin_template/tests/functional/api/test_sync.py

```diff
@@ -106,25 +113,44 @@ def test_file_decriptors(self):
         self.assertEqual(len(response), 0, response)
 
 
-class SyncInvalidURLTestCase(unittest.TestCase):
-    """Sync a repository with an invalid url on the Remote."""
+# Implement sync support before enabling this test.
+@unittest.skip("FIXME: plugin writer action required")
+class SyncInvalidTestCase(unittest.TestCase):
+    """Sync a repository with a given url on the remote."""
 
-    def test_all(self):
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+
+    def test_invalid_url(self):
+        """Sync a repository using a remote url that does not exist.
+
+        Test that we get a task failure. See :meth:`do_test`.
         """
-        Sync a repository using a Remote url that does not exist.
+        context = self.do_test('http://i-am-an-invalid-url.com/invalid/')
+        self.assertIsNotNone(context.exception.task['error']['description'])
 
-        Test that we get a task failure.
+    def test_invalid_file(self):
```

Comment:
Drop this test since it's file-plugin specific.

Reply:
I would like to keep it as a suggestion to test against broken repositories. I forgot to replace some occurences of file though.

Reply:
There's so many different ways a repository could be broken (in general) that it would probably make more sense to make a separate test class.

Reply:
Any updates on this?
ping @pulp/qe